File: LeetCode/Easy/27 RemoveElement/27RemoveElement.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] != val) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {0, 1, 2, 2, 3, 0, 4, 2};
    int val = 2;

    int k = sol.removeElement(nums, val);

    cout << "k: " << k << endl;
    cout << "Vector modificado: [";
    for (int i = 0; i < k; ++i) {
        cout << nums[i] << (i == k - 1 ? "" : ", ");
    }
    cout << "]" << endl;

    return 0;
}

File: LeetCode/Easy/27 RemoveElement/27RemoveElement.py
from typing import List

def remove_element(nums: List[int], val: int) -> int:
    k = 0 
    for i in range(len(nums)):
        if nums[i] != val:
            nums[k] = nums[i]
            k += 1
            
    return k

if __name__ == "__main__":
    lista = [0, 1, 2, 2, 3, 0, 4, 2]
    valor_a_eliminar = 2
    
    k = remove_element(lista, valor_a_eliminar)
    
    print(f"Nueva longitud k: {k}")
    print(f"Lista modificada: {lista[:k]}")